### PR TITLE
[23.0] Add option to disable the fetch_data celery task

### PR DESCRIPTION
**Pull request by jdavcs**
Ref #15700

## How to test the changes?
(Select all options that apply)
- [x] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

File: lib/galaxy/tools/execute.py [hunk 1]
```diff
@@ -180,7 +180,7 @@ def execute_single_job(execution_slice, completed_job, skip=False):
     tool_id = tool.id
     for job2 in execution_tracker.successful_jobs:
         # Put the job in the queue if tracking in memory
-        if tool_id == "__DATA_FETCH__" and tool.app.config.enable_celery_tasks:
+        if tool_id == "__DATA_FETCH__" and tool.app.config.is_fetch_with_celery_enabled:
```
Comment (mvdbeek) [inline lib/galaxy/tools/execute.py]:
```suggestion
        if tool_id == "__DATA_FETCH__" and tool.app.config.is_fetch_with_celery_enabled():
```

Comment (jdavcs) [reply lib/galaxy/tools/execute.py]: Thank you!

**Comment by bgruening** [comment]
Thanks @jdavcs, this will help us a lot.
LGTM.

File: lib/galaxy/config/__init__.py [hunk 2]
```diff
@@ -1276,6 +1277,17 @@ def check(self):
                     f"Config option '{key}' is deprecated and will be removed in a future release.  Please consult the latest version of the sample configuration file."
                 )
 
+    def is_fetch_with_celery_enabled(self):
+        """
+        True iff celery is enabled and celery_conf["task_routes"]["galaxy.fetch_data"] == DISABLED_FLAG.
```
Comment (nsoranzo) [inline lib/galaxy/config/__init__.py:1282]: @jdavcs Should this have been `celery_conf["task_routes"]["galaxy.fetch_data"] != DISABLED_FLAG` (or something like that)?

Comment (jdavcs) [reply lib/galaxy/config/__init__.py:1282]: Yes, of course! Thank you!! Will you open a PR?

Comment (nsoranzo) [reply lib/galaxy/config/__init__.py:1282]: I'll let you decide how to phrase it, if that's OK?

**Pull request merged**